packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,319 @@
+                saveResults: shouldScanTestForAccessibility,
+                testDetails: {
+                    'name': test.title,
+                    'testRunId': process.env.BS_A11Y_TEST_RUN_ID,
+                    'filePath': this._suiteFile,
+                    'scopeList': [suiteTitle, test.title]
+                },
+                platform: this._platformA11yMeta
+            }
+
+            await this.sendTestStopEvent((this._browser as WebdriverIO.Browser), dataForExtension)
+
+            if (shouldScanTestForAccessibility) {
+                log.info('Accessibility testing for this test case has ended.')
+            }
+        } catch (error) {
+            log.error(`Accessibility results could not be processed for the test case ${test.title}. Error :`, error)
+        }
+    }
+
+    /**
+      * Cucumber Only
+    */
+
+    async beforeScenario (world: ITestCaseHookParameter) {
+        if (!this._browser) {
+            return
+        }
+
+        if (!this.shouldRunTestHooks(this._browser, this._accessibility)) {
+            return
+        }
+
+        let pageOpen = true
+        const pickleData = world.pickle
+        const gherkinDocument = world.gherkinDocument
+        const featureData = gherkinDocument.feature
+        const uniqueId = getUniqueIdentifierForCucumber(world)
+        const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions)
+
+        try {
+            this._testMetadata[uniqueId] = {
+                scanTestForAccessibility : shouldScanScenario,
+                accessibilityScanStarted : true
+            }
+            const currentURL = await (this._browser as WebdriverIO.Browser).getUrl()
+            const url = new URL(currentURL)
+            pageOpen = url?.protocol === 'http:' || url?.protocol === 'https:'
+        } catch (e) {
+            pageOpen = false
+        }
+
+        if (!pageOpen) {
+            return
+        }
+
+        try {
+            if (shouldScanScenario) {
+                log.info('Setup for Accessibility testing has started. Automate test case execution will begin momentarily.')
+                await this.sendTestStartEvent(this._browser as WebdriverIO.Browser)
+            } else {
+                await this.sendTestForceStopEvent(this._browser as WebdriverIO.Browser)
+            }
+            this._testMetadata[uniqueId].accessibilityScanStarted = shouldScanScenario
+
+            if (shouldScanScenario) {
+                log.info('Automate test case execution has started.')
+            }
+        } catch (error) {
+            log.error(`Exception in starting accessibility automation scan for this test case ${error}`)
+        }
+    }
+
+    async afterScenario (world: ITestCaseHookParameter) {
+        if (!this._browser) {
+            return
+        }
+
+        if (!this.shouldRunTestHooks(this._browser, this._accessibility)) {
+            return
+        }
```

Comment:
Let's merge, I also see this multiple times now which means we should make a function so we avoid duplicated code at all cost